syne_tune/optimizer/schedulers/transfer_learning/zero_shot.py

```diff
@@ -0,0 +1,151 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+import logging
+from typing import Dict, Optional
+
+import numpy as np
+import pandas as pd
+import xgboost
+
+from benchmarking.blackbox_repository.blackbox_surrogate import BlackboxSurrogate
+from syne_tune.config_space import Domain
+from syne_tune.optimizer.schedulers.searchers.searcher import BaseSearcher
+from syne_tune.optimizer.schedulers.transfer_learning import TransferLearningTaskEvaluations, TransferLearningMixin
+
+logger = logging.getLogger(__name__)
+
+
+class ZeroShotTransfer(TransferLearningMixin, BaseSearcher):
+    def __init__(
+            self,
+            config_space: Dict,
+            transfer_learning_evaluations: Dict[str, TransferLearningTaskEvaluations],
+            metric: str,
+            mode: str = 'min',
+            sort_transfer_learning_evaluations: bool = True,
+            use_surrogates: bool = False,
+            random_seed: Optional[int] = None
+    ) -> None:
+        """
+        A zero-shot transfer hyperparameter optimization method which selects configurations that minimize the average
+        rank obtained on historic metadata (transfer_learning_evaluations).
+
+        Reference: Sequential Model-Free Hyperparameter Tuning.
+        Martin Wistuba, Nicolas Schilling, Lars Schmidt-Thieme.
+        IEEE International Conference on Data Mining (ICDM) 2015.
+
+        :param config_space: Configuration space for trial evaluation function.
+        :param transfer_learning_evaluations: Dictionary from task name to offline evaluations.
+        :param metric: Objective name to optimize, must be present in transfer learning evaluations.
+        :param mode: Whether to minimize (min) or maximize (max)
+        :param sort_transfer_learning_evaluations: Use False if the hyperparameters for each task in
+        transfer_learning_evaluations Are already in the same order. If set to True, hyperparameters are sorted.
+        :param use_surrogates: If the same configuration is not evaluated on all tasks, set this to true. This will
```

Comment:
Couldnt this one be inferred automatically? (that configurations are not all the same in all tasks)

Reply:
In term of runtime complexity, if you have n configs and k datasets, it would require `O(k n log n)` which does not sound too bad right? Probably less than a few seconds in the largest of our datasets I guess.

I leave you the choice so it is not a blocking comment :) 
My only important points are on adding comments.